contracts/market-wrapper/NounsMarketWrapper.sol

```diff
@@ -0,0 +1,133 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.5;
+
+// ============ External Imports ============
+import {INounsAuctionHouse} from "../external/interfaces/INounsAuctionHouse.sol";
+
+// ============ Internal Imports ============
+import {IMarketWrapper} from "./IMarketWrapper.sol";
+
+/**
+ * @title NounsMarketWrapper
+ * @author Anna Carroll + Nounders
+ * @notice MarketWrapper contract implementing IMarketWrapper interface
+ * according to the logic of Nouns' Auction Houses
+ */
+contract NounsMarketWrapper is IMarketWrapper {
+    // ============ Internal Immutables ============
+
+    INounsAuctionHouse internal immutable market;
+
+    // ======== Constructor =========
+
+    constructor(address _nounsAuctionHouse) {
+        market = INounsAuctionHouse(_nounsAuctionHouse);
+    }
+
+    // ======== External Functions =========
+
+    /**
+     * @notice Determine whether there is an existing auction
+     * for this token is active
+     * @return TRUE if the auction exists
+     */
+    function auctionExists(uint256 auctionId)
+      public
+      view
+      override
+      returns (bool)
+    {
+        (uint256 currentAuctionId, , , , , ) = market.auction();
+        return auctionId == currentAuctionId;
+    }
+
+    /**
+     * @notice Determine whether the given auctionId and tokenId is active.
+     * We ignore nftContract since it is static for all nouns auctions.
+     * @return TRUE if the auctionId and tokenId matches the active auction
+     */
+    function auctionIdMatchesToken(
+        uint256 auctionId,
+        address /* nftContract */,
+        uint256 tokenId
+    ) public view override returns (bool) {
+        (uint256 currentAuctionId, , , , , ) = market.auction();
+        return currentAuctionId == tokenId && currentAuctionId == auctionId;
+    }
+
+    /**
+     * @notice Calculate the minimum next bid for the active auction
+     * @return minimum bid amount
+     */
+    function getMinimumBid(uint256 /* auctionId */)
```

Comment:
i'm wondering if it'd be good to require that the `auctionId` passed in as the param matches the current auction's `nounId` -- in this and other functions -- so we fail hard if we somehow ever accidentally ask for the minimumBid or getCurrentHighestBidder for an old token id

Reply:
Thought about the same, but the checks in PartyBid's `bid` function wouldn't allow it to get this far. It would never pass the `!isFinalized` check.

Reply:
maybe i'm just paranoid, but i'd still like to see these require statements in, just to prevent something unexpected happening in a future iteration of the `bid` function.